rpc/handlers.go

```diff
@@ -479,11 +481,101 @@ func (h *Handler) Nonce(id *BlockID, address *felt.Felt) (*felt.Felt, *jsonrpc.E
 	}
 	if err != nil {
 		return nil, ErrContractNotFound
 	}
 
 	return nonce, nil
 }
+
+// https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L248
+func (h *Handler) Class(id *BlockID, classHash *felt.Felt) (*Class, *jsonrpc.Error) {
+	state, stateCloser, err := h.stateByBlockID(id)
+	if err != nil {
+		return nil, ErrBlockNotFound
+	}
+	declared, err := state.Class(classHash)
+	if closerErr := stateCloser(); closerErr != nil {
+		h.log.Errorw("Error closing state reader in getClass", "err", closerErr)
+	}
+
+	if err != nil {
+		return nil, ErrClassHashNotFound
+	}
+
+	var rpcClass *Class
+	switch c := declared.Class.(type) {
+	case *core.Cairo0Class:
+		rpcClass = &Class{
+			Abi:         c.Abi,
+			Program:     c.Program,
+			EntryPoints: EntryPoints{},
+		}
+
+		rpcClass.EntryPoints.Constructor = make([]EntryPoint, 0, len(c.Constructors))
+		for _, entryPoint := range c.Constructors {
+			rpcClass.EntryPoints.Constructor = append(rpcClass.EntryPoints.Constructor, EntryPoint{
+				Offset:   entryPoint.Offset,
+				Selector: entryPoint.Selector,
+			})
+		}
+
+		rpcClass.EntryPoints.L1Handler = make([]EntryPoint, 0, len(c.L1Handlers))
+		for _, entryPoint := range c.L1Handlers {
+			rpcClass.EntryPoints.L1Handler = append(rpcClass.EntryPoints.L1Handler, EntryPoint{
+				Offset:   entryPoint.Offset,
+				Selector: entryPoint.Selector,
+			})
+		}
+
+		rpcClass.EntryPoints.External = make([]EntryPoint, 0, len(c.Externals))
+		for _, entryPoint := range c.Externals {
+			rpcClass.EntryPoints.External = append(rpcClass.EntryPoints.External, EntryPoint{
+				Offset:   entryPoint.Offset,
+				Selector: entryPoint.Selector,
+			})
+		}
+
+	case *core.Cairo1Class:
+		rpcClass = &Class{
+			Abi:                  c.Abi,
+			SierraProgram:        c.Program,
+			ContractClassVersion: c.SemanticVersion,
+			EntryPoints:          EntryPoints{},
+		}
+
+		rpcClass.EntryPoints.Constructor = make([]EntryPoint, 0, len(c.EntryPoints.Constructor))
+		for _, entryPoint := range c.EntryPoints.Constructor {
+			index := entryPoint.Index
+			rpcClass.EntryPoints.Constructor = append(rpcClass.EntryPoints.Constructor, EntryPoint{
+				Index:    &index,
+				Selector: entryPoint.Selector,
+			})
+		}
+
+		rpcClass.EntryPoints.L1Handler = make([]EntryPoint, 0, len(c.EntryPoints.L1Handler))
+		for _, entryPoint := range c.EntryPoints.L1Handler {
+			index := entryPoint.Index
+			rpcClass.EntryPoints.L1Handler = append(rpcClass.EntryPoints.L1Handler, EntryPoint{
+				Index:    &index,
+				Selector: entryPoint.Selector,
+			})
+		}
+
+		rpcClass.EntryPoints.External = make([]EntryPoint, 0, len(c.EntryPoints.External))
+		for _, entryPoint := range c.EntryPoints.External {
+			index := entryPoint.Index
+			rpcClass.EntryPoints.External = append(rpcClass.EntryPoints.External, EntryPoint{
+				Index:    &index,
+				Selector: entryPoint.Selector,
+			})
+		}
+
+	default:
+		return nil, ErrInternalError
```

Comment:
Not sure we should have an internalError since the spec doesn't define it. We should probably just return `ErrClassHashNotFound` in this case. Maybe log errors so we can track these cases?
```suggestion
		return nil, ErrClassHashNotFound
```